### ISSUE TITLE
Using dash-leaflet map inside tab components (v0.1.7)

### ISSUE THREAD

**Issue by ZhabizGIT** [thedirtyfew/dash-leaflet#56]
I have recently upgraded the dash-leaflet library from v0.0.22 to v0.1.7. I was using two maps to show separately within tab sections. But after this upgrade, I cannot use them anymore. Because when I move the cursor from the first tab to the second and get back to the first tab, the map is disappeared. the following error message is logged in the browser console. In the older version of dash-leaflet v0.0.22, I would not run into this issue.

**Error message:** "TypeError: Cannot read property 'leafletElement' of null"

The sample code to describe this scenario is as follows:

```
import dash
import dash_leaflet as dl
import dash_core_components as dcc
import dash_html_components as html

app = dash.Dash()

app.layout = html.Div([
    dcc.Tabs([
        dcc.Tab(label='Tab one', children=[
            dl.Map(id="map1",children=[dl.TileLayer(id="tile1")], style={'width': '1000px', 'height': '500px'})
        ]),
        dcc.Tab(label='Tab two', children=[
            dl.Map(id="map2",children=[dl.TileLayer(id="tile2")],center=[40, -100], zoom=4, style={'width': '1000px', 'height': '500px'})
        ]),
    ])
])


if __name__ == '__main__':
    app.run_server(port=8030)

```
In the initial loading of the page, we have the following result:
![initial_layout](https://user-images.githubusercontent.com/18639751/99321851-c00a9580-283c-11eb-8a83-621f7ee9834f.png)

After the first round of switching between tabs, we have empty bodies which come up with error messages:
![error](https://user-images.githubusercontent.com/18639751/99322302-ae75bd80-283d-11eb-972d-f9224840b827.png)

Please let me know what can be done to fix the issue. I prefer not to downgrade the dash-leaflet.

**Comment by dmasip-rxd** [comment]
Same issue here, have you tried with newer versions than `0.0.22`?

**Comment by mirkomiorelli** [comment]
Same for me.

**Comment by emilhe** [comment]
Ah, sorry, that was an introduced bug. It's due to a new feature that tracks the view port of the map on init; when you switch tabs the check is performed before the map is ready (in seem that in this case, react re-creates the component). I have added a check if the map is ready, which should solve the issue. I have pushed the update in the 0.1.8 release,

https://pypi.org/project/dash-leaflet/0.1.8/

**Comment by ZhabizGIT** [comment]
Thank you @emilhe for your time to handle this issue so quickly. It is almost fixed. But there are some small functional issues:

1. After updating the package to the latest version, I noticed we sometimes need to switch between tabs twice to see the map in the second tabbed section. 

2. The switching between tabs reset any changes on the maps such as drawn measurement shapes and zooming, ... to the default view. 
 
I really appreciated it if you could take a look and fix them at your convenience.

Thank you!

**Comment by emilhe** [comment]
Do you remember if these issues are "new", i.e. were then present altso in 0.0.22?

**Comment by ZhabizGIT** [comment]
Yes, that is true. I tested that as well. Using v0.0.22, we can smoothly switch between tabs. The zooming changes and center movements on the maps are kept after switching between tabs. In v0.1.8, a map with the default center (the lat and lon of Denmark I guess) is displayed sometimes. Here is the code:

```
import dash
import dash_leaflet as dl
import dash_core_components as dcc
import dash_html_components as html

app = dash.Dash()

app.layout = html.Div([
    dcc.Tabs(value="tab1",children=[
        dcc.Tab(value="tab1",label='Tab one', children=[
            dl.Map(id="map1",children=[dl.TileLayer(id="tile1"),
                                       dl.MeasureControl(position="topleft", primaryLengthUnit="kilometers", primaryAreaUnit="hectares",
                                        activeColor="#214097", completedColor="#972158")
                                       ],zoom=4, center=(52.0, -11.0), style={'width': '1000px', 'height': '500px'}),
            html.P("tab1")
        ]),
        dcc.Tab(value="tab2",label='Tab two', children=[
            dl.Map(id="map2",children=[dl.TileLayer(id="tile2"),
                                       dl.MeasureControl(position="topleft", primaryLengthUnit="kilometers", primaryAreaUnit="hectares",
                                        activeColor="#214097", completedColor="#972158")
                                        ],center=[40, -100], zoom=4, style={'width': '1000px', 'height': '500px'}),
                html.P("tab2")
        ]),
    ])
])


if __name__ == '__main__':
    app.run_server(port=8030)
```

The MeasureControl has been recently added. I don't know if it is possible to test that in older versions.

**Comment by ZhabizGIT** [comment]
As a temporary solution, I downgraded the package to v0.1.4. The issue that changes the center of maps to the default coordinate (the latitude and longitude of Denmark) when we switch between tabs is fixed!

**Comment by emilhe** [comment]
Thanks for pointing me to 0.1.4, that help my locate the root cause of the issue. It should now be fixed in 0.1.11 :)

https://pypi.org/project/dash-leaflet/
